burn-tensor/src/tensor/api/float.rs

```diff
@@ -103,6 +104,24 @@ where
         Self::from_data(floats.into().convert())
     }
 
+    /// Returns a new tensor with the same shape and device as the current tensor and the data
+    /// casted to Integer.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::Tensor;
+    ///
+    /// fn example<B: Backend>() {
+    ///     let float_tensor = Tensor::<B, 1>::from_floats([1.0, 2.0]);
+    ///     let int_tensor = float_tensor.int();
+    /// }
+    /// ```
+    pub fn int(&self) -> Tensor<B, D, Int> {
+        Tensor::<B, D, Int>::from_data(self.to_data().convert())
+    }
```

Comment:
I would take ownership here and call `into_data`. In a following PR we should call a backend function where each backend can decide how they do the conversion. For instance, in this current implementation, the wgpu backend would need to allocate at least 3 new buffers, where 0 or 1 could have been allocated with proper casting.